release_dashboard/lib/logic/git.dart

```diff
@@ -47,3 +47,63 @@ class git {
     }
   }
 }
+
+abstract class gitValidation {
+  /// Returns the regex needed to validate this type of input.
+  RegExp get regex;
+
+  /// Returns the error message if the regex validation fails.
+  String get errorMsg;
+
+  /// Uses [regex] to validate the input provided. Returns true if it is valid, false otherwise.
+  bool validate(String? input);
+}
+
+/// Provides all the tools and methods to validate a single Git hash such as the dart revision.
+class gitHash extends gitValidation {
+  // Allows empty string because the dart revision which is the only input using this regex could be empty
+  final RegExp gitHashRegex = RegExp(r'^$|^[0-9a-f]{40}$');
+  final String gitHashErrorMsg = 'Must be 40 alphanumeric characters or an empty string.';
+
+  RegExp get regex => gitHashRegex;
+
+  String get errorMsg => gitHashErrorMsg;
+
+  bool validate(String? input) {
+    return this.regex.hasMatch(input == null ? '' : input.trim()); // removes leading or trailing whitespaces
+  }
+}
+
+/// Provides all the tools and methods to validate a multiple Git hash entry such as cherrypicks.
+class multiGitHash extends gitValidation {
+  /// Supports multiple git hashes delimited by a comma or an empty string.
+  ///
+  /// valid: c7714158950347,c7714158950347,c7714158950347
+  /// valid: c7714158950347
+  /// invalid:   c7714158950347,@@cccccc
+  /// invalid (cannot end with a comma):   c7714158950347,c7714158950347,
+  final RegExp multiGitHashRegex = RegExp(r'^$|^[0-9a-z]{40}(?:,[0-9a-z]{40})*$');
```

Comment:
Instead of a multi git hash, we could simplify to `Iterable<bool> hashes = gitHashes.split(',').map((String gitHash) => GitHash.validate(githash));`.

Regexes are pretty much their own programming language, and should be avoided where possible.